$OpenBSD: patch-libavcodec_aacenc_tns_h,v 1.2 2015/10/25 08:46:41 ajacoutot Exp $

aacenc_tns: redo coefficient quantization and decision making

aacenc_tns: readjust values for new TNS decision making

acenc_tns: rework TNS descision logic

--- libavcodec/aacenc_tns.h.orig	Mon Sep  7 21:58:01 2015
+++ libavcodec/aacenc_tns.h	Sun Oct 18 22:26:54 2015
@@ -30,21 +30,6 @@
 
 #include "aacenc.h"
 
-/* Could be set to 3 to save an additional bit at the cost of little quality */
-#define TNS_Q_BITS 4
-
-/* TNS will only be used if the LPC gain is within these margins */
-#define TNS_GAIN_THRESHOLD_LOW  1.395f
-#define TNS_GAIN_THRESHOLD_HIGH 11.19f
-
-/* If the energy ratio between the low SFBs vs the high SFBs is not between
- * those two values, use 2 filters instead */
-#define TNS_E_RATIO_LOW  0.77
-#define TNS_E_RATIO_HIGH 1.23
-
-/* Do not use TNS if the psy band spread is below this value */
-#define TNS_SPREAD_THRESHOLD 37.081512f
-
 void ff_aac_encode_tns_info(AACEncContext *s, SingleChannelElement *sce);
 void ff_aac_apply_tns(AACEncContext *s, SingleChannelElement *sce);
 void ff_aac_search_for_tns(AACEncContext *s, SingleChannelElement *sce);
